Feature/admin edit periods

usu opened the pull request:
- Kompenenten in kleinere Stücke gesplittet
- Input-Daten (Kontext wie camp, period, etc.) von Anzeige/visibility getrennt
- Nutzung von v-dialog slot "activator"
- kleinere Bugs/Verbesserungen

usu committed:
move component API logic to mixin + snapshot tests

carlobeltrame committed:
Update dependency to fix audit warning

carlobeltrame committed:
Update hal-json-normalizer to enable embedded collections with self link

carlobeltrame committed:
Re-add events link to camp and use it to fetch all events at once for the picasso

carlobeltrame committed:
Visual improvements

carlobeltrame committed:
Update frontend dependencies to the latest versions

Also remove git from the frontend container, because we now have no
dependencies from GitHub anymore again.

carlobeltrame committed:
Fix linter errors

carlobeltrame committed:
Update remaining outdated frontend dependencies

usu committed:
split into smaller components; no other code change

usu committed:
rework props and visibility handling of dialogs

usu committed:
use v-slot:activator for v-dialog; eliminate need for v-model on top-level

manuelmeister committed:
Merge pull request #375 from carlobeltrame/restore-events-link

Restore events link on camp

carlobeltrame commented on frontend/src/components/base/ContentCard.vue:
```diff
@@ -0,0 +1,36 @@
+<!--
+Displays details on a single camp and allows to edit them.
```
```suggestion
Displays the content wrapped inside a card.
```

usu replied:
Sehr gut. Hab ich mal noch daran gedacht die Comments anzupassen und dann vergessen. Mach ich grad überall 👍

carlobeltrame commented on frontend/src/components/camp/CampPeriods.vue:
```diff
@@ -0,0 +1,61 @@
+<!--
+Displays details on a single camp and allows to edit them.
```
```suggestion
Displays the periods of a single camp and allows to edit them.
```

carlobeltrame commented on frontend/src/components/camp/CampPeriodsListItem.vue:
```diff
@@ -0,0 +1,70 @@
+<!--
+Displays details on a single camp and allows to edit them.
```
```suggestion
Displays a period as a list item including controls to edit and delete it.
```

carlobeltrame commented on frontend/src/views/camp/CampHome.vue:
```diff
@@ -0,0 +1,31 @@
+<!--
+Displays details on a single camp and allows to edit them.
```
```suggestion
Displays details and periods of a single camp and allows to edit them.
```

carlobeltrame commented on frontend/src/components/dialog/CreatePeriodDialog.vue:
```diff
@@ -1,12 +1,14 @@
     submit-color="success"
-    :cancel-action="close">
+    :cancel-action="close"
+    :value="value"
```
Ist diese Zeile nicht schon automatisch in `v-bind="$attrs"` impliziert?

usu replied:
Ist nach dem dritten Commit wieder mit `v-model="showDialog"` umgesetzt und daher eh weg. Ich hatte allerdings manchmal Mühe, dass value mit $attrs nicht mitgekommen ist. Wieso genau bin ich aber nicht auf die Schliche gekommen. Habe dann bei Codes von anderen gesehen, dass sie `:value`  noch explizit listen.

carlobeltrame commented on frontend/src/components/dialog/DialogBase.vue:
```diff
@@ -2,7 +2,7 @@
   name: 'DialogBase',
   props: {
-    value: { default: null, required: true }
+    value: { type: Boolean, default: false, required: true }
```
Macht ein default Sinn wenn die prop required ist?

usu replied:
Gibt's im DialogBase nicht mehr aber im DialogForm. Hab dort `default` entfernt.

carlobeltrame commented on frontend/src/components/dialog/DialogForm.vue:
```diff
@@ -1,7 +1,9 @@
   <v-dialog
     v-bind="$attrs"
-    v-on="$listeners">
+    :value="value"
```
Auch hier, braucht es diese Zeile wenn sowieso schon `v-bind="$attrs"` dasteht?

usu replied:
Funktioniert für mich nicht, `value` ist nicht Teil von `$attr`. Bin auf Google nicht schlüssig geworden, ob das so gedacht ist oder nicht.

carlobeltrame commented on frontend/src/components/dialog/DialogForm.vue:
```diff
@@ -1,7 +1,9 @@
-    v-on="$listeners">
+    :value="value"
+    v-on="$listeners"
+    @input="$emit('input', $event)">
```
Dasselbe mit dieser Zeile und `v-on="$listeners"`?

usu replied:
Tatsächlich, das geht. Wieder was dazu gelernt.

carlobeltrame commented on frontend/src/components/dialog/EditPeriodDialog.vue:
```diff
@@ -1,11 +1,13 @@
-    :cancel-action="close">
+    :cancel-action="close"
+    :value="value"
+    v-bind="$attrs"
```
Und hier nochmals

usu replied:
Ist dort jetzt eh weg

carlobeltrame commented on frontend/src/components/camp/CampPeriodsListItem.vue:
```diff
@@ -11,35 +11,21 @@ Displays details on a single camp and allows to edit them.
+        </edit-period-dialog>
+
+        <delete-entity-dialog v-model="showDeleteDialog" :entity="period">
+          <template v-slot:activator="{ on }">
+            <button-delete v-on="on" />
+          </template>
```
Mein Kopf schmerzt... :dizzy_face: :wink:

usu replied:
Man gewöhnt sich daran 😄

carlobeltrame commented on frontend/src/components/camp/CampPeriodsListItem.vue:
```diff
@@ -11,35 +11,21 @@ Displays details on a single camp and allows to edit them.
+          <template v-slot:activator="{ on }">
+            <button-delete v-on="on" />
+          </template>
+          the Period "{{ period.description }}"
```
:+1:

carlobeltrame commented on frontend/src/components/camp/CampPeriodsListItem.vue:
```diff
@@ -11,35 +11,21 @@ Displays details on a single camp and allows to edit them.
-    <edit-period-dialog v-model="showEditDialog" :period="period" />
     <delete-entity-dialog v-model="showDeleteDialog" :entity="period">
       the Period "{{ period.description }}"
     </delete-entity-dialog>
```
Bitte entfernen, denke das ist nicht mehr nötig oder?

usu replied:
Yes, danke

usu committed:
small changes as per review

usu committed:
encapsulate API logic into wrapper component

usu committed:
add Checkbox, ColorPicker, DatePicker, TimePicker, Textarea

usu committed:
Merge branch 'devel' into feature/api-input

usu committed:
initial test script for ApiWrapper

usu committed:
input/update test for ApiTextField

usu committed:
refactor test setup for ApiWrapper & ApiTextField

carlobeltrame committed:
Don't attempt to reload entities that are already marked as deleting

Before this change, when deleting an entity which has a circular
reference to another entity (e.g. a camp and its owner, or a period and a
day in it), we would reload both of them indefinitely, always getting
status 404. We can avoid this by excluding entities from the reloading
if they are already marked as deleting.

carlobeltrame committed:
Make sure the circular dependency breaking also works when the delete is triggered from outside the cycle

usu committed:
changes as per review: general look & feel; picker improvements

usu committed:
refactor BasePicker

carlobeltrame committed:
Merge pull request #379 from carlobeltrame/fix-circular-reference-deleting-bug

Während dem Löschvorgang keine Entities neu laden die als deleting markiert sind

usu committed:
rename SingleSelect --> Select

usu committed:
remove 'editing' prop; accept standard 'readonly' and 'disabled' prop

usu committed:
Migrate from travis.org to travis.com

usu committed:
smaller fixes as per review

usu committed:
Merge pull request #372 from usu/feature/api-input

API user inputs

usu committed:
merge devel into feature/admin-edit-periods

usu committed:
small fixes as per review; renaming of dialogs
